model-engine/model_engine_server/api/llms_v1.py

```diff
@@ -71,6 +75,36 @@
 logger = make_logger(filename_wo_ext(__name__))
 
 
+def handle_streaming_exception(
+    e: Exception,
+    code: int,
+    message: str,
+):
+    tb_str = traceback.format_exception(etype=type(e), value=e, tb=e.__traceback__)
+    request_id = get_request_id()
+    timestamp = datetime.now(pytz.timezone("US/Pacific")).strftime("%Y-%m-%d %H:%M:%S %Z")
+    structured_log = {
+        "error": message,
+        "request_id": str(request_id),
+        "traceback": "".join(tb_str),
+    }
+    logger.error("Exception: %s", structured_log)
+    return {
+        "data": json.dumps(
+            {
+                "request_id": str(request_id),
+                "error": {
+                    "status_code": code,
```

Comment:
Maybe this is a bit pedantic (pydantic? lol), but since you ended up creating the DTOS anyway, wonder if it makes sense to instantiate the DTOs and convert them to JSON? That way you get the typing.

Alternatively, I think we want a unit test to enforce the API contract behind the error return type.

Reply:
will do. for test i added `test_completion_stream_endpoint_not_found_returns_404` but unfortunately i still haven't figured out how to run it